### DISCUSSION TITLE
Wanting to use your module inside an other one

### DISCUSSION THREAD

**Discussion by adrien-schiehle** [ChadKeating/rolladvantage-token-stamp-2-foundry#12] [General]
Greetings,

I'm currently working on the world-anvil integration module inside Foundy VTT :  [core module](https://gitlab.com/foundrynet/world-anvil)

One of the features I'm trying to add is : Automatic actor creation when importing character article from world-anvil.

My current work is here : [additional actor module](https://gitlab.com/adrien.schiehle/world-anvil-actors)  (for automatic actor creation)

I've stumble into a small issue : 
- When importing articles from world-anvil, related images are kept inside world-anvil site. When displaying images, fvtt is referencing the foreign address.
- It works well from Journal Entry images and main Actor images.
- But for tokens, it triggers CORS problems

So I thought : Why not generating token images on the fly when creating the Actor ?

Since I'm personally using [your website](https://rolladvantage.com/tokenstamp/) for my characters, I wish to use your foundry module inside my own. (meaning adding it as a dependency)

Is it something you would allow ?

If yes, I may have some question on how to cleanly integrate it.

**Comment by RollAdvantage** [comment]
Hey,

Yeah go ahead use this as a dependancy. Although I dont know how much help I'll be. I dont work with js that much and ts2 isnt built in anyway to handle pragmatically generating tokens. 

Though I might suggest posting an issue about CORs on the world anvil module. I believe the two have a special relationship hence the integration so it might be an oversight that could be addressed.

There also might be a way around it by loading an image into a canvas, exporting it to a data url and saving it as a png with foundry. (if it works for journal entries save that image again)

Thanks,
Chad

**Reply by adrien-schiehle** [reply]
Thanks for the tips.
I will start by asking Athropos and Dimitris then.
